posthog/temporal/batch_exports/batch_exports.py

```diff
@@ -426,6 +465,55 @@ async def finish_batch_export_run(inputs: FinishBatchExportRunInputs) -> None:
         )
 
 
+async def pause_batch_export_if_over_failure_threshold(
+    batch_export_id: str, check_window: int, failure_threshold: int = 10
+) -> bool:
+    """Pause a batch export if it exceeds failure threshold.
+
+    A 'check_window' was added to account for batch exports that have a history of failures but have some
+    occassional successes in the middle. This is relevant particularly for low-volume exports:
+    A batch export without rows to export always succeeds, even if it's not properly configured. So, the failures
+    could be scattered between these successes.
+
+    Keep in mind that if 'check_window' is less than 'failure_threshold', there is no point in even counting,
+    so we immediately return.
+
+    We check if the count of failed runs in the last 'check_window' runs exceeds 'failure_threshold'. This means
+    that 'pause_batch_export_if_over_failure_threshold' should only be called when handling a failed run,
+    otherwise we could be pausing a batch export that is just now recovering (as old failed runs in 'check_window'
+    contribute to exceeding 'failure_threshold').
+
+    Arguments:
+        batch_export_id: The ID of the batch export to check and pause.
+        check_window: The window of runs to consider for computing a count of failures.
+        failure_threshold: The number of runs that must have failed for a batch export to be paused.
+
+    Returns:
+        A bool indicating if the batch export is paused.
+    """
+    if check_window < failure_threshold:
+        return False
```

Comment:
Probably minor, but would this be a programmer error? Why would we want to hit this branch?

Reply:
Yeah, always a programmer error, could raise an exception here instead.